mapper/orm/src/main/java/org/hibernate/search/mapper/orm/search/query/SearchQuery.java

```diff
@@ -14,22 +14,102 @@
+	 */
 	List<T> getResultList();
 
+	/**
+	 * Execute the query and retrieve the results as a single element.
+	 *
+	 * @return The result.
+	 * @throws org.hibernate.search.util.common.SearchException If something goes wrong while executing the query.
+	 * @throws org.hibernate.HibernateException If something goes wrong while fetching results from the database,
+	 * or the number of results is not exactly one.
+	 * @throws javax.persistence.PersistenceException If something goes wrong while fetching results from the database,
+	 * or the number of results is not exactly one.
+	 */
 	T getSingleResult();
 
+	/**
+	 * Execute the query and retrieve the results as a single, optional element.
+	 *
+	 * @return The result.
+	 * @throws org.hibernate.search.util.common.SearchException If something goes wrong while executing the query.
+	 * @throws org.hibernate.HibernateException If something goes wrong while fetching results from the database,
+	 * or the number of results is more than one.
+	 * @throws javax.persistence.PersistenceException If something goes wrong while fetching results from the database,
+	 * or the number of results is more than one.
+	 */
 	Optional<T> getOptionalResult();
 
+	/**
+	 * Execute the query and retrieve the total result size,
+	 * ignoring pagination settings ({@link #setMaxResults(int)} and {@link #setFirstResult(int)}).
+	 *
+	 * @return The total number of results, ignoring pagination settings.
+	 * @throws org.hibernate.search.util.common.SearchException If something goes wrong while executing the query.
+	 */
 	long getResultSize();
 
+	/**
+	 * Set the maximum number of results returned by this query.
+	 * <p>
+	 * The default is no limit.
+	 *
+	 * @param maxResults The maximum number of results to return. Must be positive or zero.
+	 * @return {@code this} for method chaining.
+	 */
 	SearchQuery<T> setMaxResults(int maxResults);
 
+	/**
+	 * Set the offset of the first result returned by this query.
+	 * <p>
+	 * The default offset is {@code 0}.
+	 *
+	 * @param firstResult The offset of the first result. Must be positive or zero.
+	 * @return {@code this} for method chaining.
+	 */
 	SearchQuery<T> setFirstResult(int firstResult);
 
+	/**
+	 * Set the JDBC fetch size for this query.
+	 *
+	 * @param fetchSize The fetch size. Must be positive or zero.
+	 * @return {@code this} for method chaining.
+	 * @see Query#setFetchSize(int)
+	 */
 	SearchQuery<T> setFetchSize(int fetchSize);
 
+	/**
+	 * Convert this query to a {@link TypedQuery JPA query}.
+	 * <p>
+	 * Note that the resulting query <strong>does not support all operations</strong>
+	 * and may behave slightly differently in some cases
+	 * (including, but not limited to, the type of thrown exceptions).
+	 * For these reasons, it is recommended to only use this method
+	 * when integrating to an external library that expects JPA queries.
+	 *
+	 * @return A representation of this query as a JPA query.
```

Comment:
I'm not sure I'm following, we did agree it was useful to be able to use Hibernate Search queries through JPA/ORM interfaces, didn't we? I just wanted to move this away from our main interfaces because we know that it will always be just a hack and we cannot implement all the methods of `TypedQuery`.

It's not a JPA query, it's a representation of a Hibernate Search query as a JPA query. The only purpose of this method is to allow using JPA interfaces to manipulate Hibernate Search queries, thereby integrating more easily into existing frameworks that are used to Hibernate ORM.